src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public OSName(string name)
+        {
+            _name = name;
+        }
+
+        public static OSName Windows { get { return new OSName(WindowsName); } }
+        public static OSName Linux { get { return new OSName(LinuxName); } }
+        public static OSName OSX { get { return new OSName(OSXName); } }
+
+        public bool Equals(OSName osName)
+        {
+            return (string.IsNullOrEmpty(_name) && string.IsNullOrEmpty(osName._name)) ?
```

Comment:
I would think you'd want to parameter validate the constructor, preventing null and empty. And then you'd only have to deal with the null case internally for null when `default(OSName)` is used.